src/clr-angular/forms/radio/radio.ts

```diff
@@ -4,17 +4,48 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { Directive, ViewContainerRef } from '@angular/core';
+import { Directive, ElementRef, HostListener, OnInit, Optional, Renderer2, ViewContainerRef } from '@angular/core';
+import { NgControl } from '@angular/forms';
 
+import { IfErrorService } from '../common/if-error/if-error.service';
+import { ControlClassService } from '../common/providers/control-class.service';
+import { NgControlService } from '../common/providers/ng-control.service';
 import { WrappedFormControl } from '../common/wrapped-control';
-
-import { ClrRadioContainer } from './radio-container';
+import { ClrRadioWrapper } from '../radio/radio-wrapper';
 
 @Directive({ selector: '[clrRadio]' })
-export class ClrRadio extends WrappedFormControl<ClrRadioContainer> {
-  // Once again, several more elegant solutions were foiled by severity 3+ bugs on Angular that have been opened
-  // for 6 months to a year. So that's how we do it. Inheritance and ridiculous constructors. :-(
-  constructor(vcr: ViewContainerRef) {
-    super(ClrRadioContainer, vcr);
+export class ClrRadio extends WrappedFormControl<ClrRadioWrapper> implements OnInit {
+  constructor(
+    vcr: ViewContainerRef,
+    @Optional() private ngControlService: NgControlService,
+    @Optional() private ifErrorService: IfErrorService,
+    @Optional() private control: NgControl,
+    @Optional() controlClassService: ControlClassService,
+    renderer: Renderer2,
+    el: ElementRef
+  ) {
+    super(ClrRadioWrapper, vcr, 0);
+    if (!control) {
+      throw new Error(
+        'clrRadio can only be used within an Angular form control, add ngModel or formControl to the input'
```

Comment:
The hard dependency on Angular forms is actually a problem for Radio and Checkbox. Because they are both entirely rendered in CSS, they need empty labels, matching `id` and `for` attributes, etc., to display correctly. This is a nightmare when generating dynamically many of them for non-form purposes (lists, trees, datagrids, ...) if you **don't** want to import `FormsModule`, which is a perfectly reasonable request.

The thing to note is that in these cases, there is no validation. No helper text. So I believe we really want to make an exception for radios and checkboxes and support them outside of Angular forms too.

We added the Angular components at the time so that people